### PR TITLE
Support multi-dimensional output for CP regression

### LINKED ISSUE #51: Tensor regression with `y` of arbitrary order
Right now using something like 
```python
estimator = TuckerRegressor(weight_ranks=[5], tol=10e-7, n_iter_max=100, reg_W=1, verbose=0)
estimator.fit(X,y)
```
demands that `X` be of dimensions (num_samples, num_features) and `y` be of dimension (num_samples). Why can't X and y be arbitrary tensor dimensions? For example, if I want to find a weight tensor that will minimize the error between an image (num_images, width, height, channels) and  its rotated version (num_images, width, height, channels), it doesn't appear I can do that with the tensor regression API. Or if I want to learn a tensor that will contract with a vector to produce some simple 2d image (i.e. a matrix).

### PR DESCRIPTION
**Pull request by merajhashemi**
This pr resolves #51 (CP only).

### CODE REVIEW

**Comment by merajhashemi** [comment]
I will add new test cases for multi-dimensional outputs🙂

File: tensorly/regression/cp_regression.py [hunk 1]
```diff
@@ -80,12 +82,13 @@ def fit(self, X, y):
...
         rng = T.check_random_state(self.random_state)
 
-        # Initialise randomly the weights
+        # Initialise the weights randomly
         W = []
+
         for i in range(
             1, T.ndim(X)
         ):  # The first dimension of X is the number of samples
-            W.append(T.tensor(rng.randn(X.shape[i], self.weight_rank), **T.context(X)))
```
Comment (aarmey) [inline tensorly/regression/cp_regression.py:88]: Check that I resolved the merge conflict correctly here.

**Review by aarmey — CHANGES_REQUESTED** [review]
This looks great @merajhashemi! Could you please run black formatting on the code? `black tensorly` in the top directory should do it.

File: tensorly/regression/cp_regression.py (hunk 1 above)
Comment (aarmey) [reply tensorly/regression/cp_regression.py:88]: Indeed, I definitely think I messed up the merge here.

**Comment by JeanKossaifi** [comment]
@merajhashemi if you have the time would be great to actually get this merged! :)

File: tensorly/regression/cp_regression.py (hunk 1 above)
Comment (merajhashemi) [reply tensorly/regression/cp_regression.py:87]: I fixed it. Everything is working fine 😀

**Comment by merajhashemi** [comment]
@JeanKossaifi I'm not sure what was left from this PR. The tests were passing; I just had to update the code style. I think it's ready to be merged 😄

**Comment by aarmey** [comment]
Thanks @merajhashemi! Merging.

**Comment by JeanKossaifi** [comment]
Fabulous, thank you @merajhashemi!

**Pull request merged**